examples/golang-todo-rest-crud/cmd/todo/main.go

```diff
@@ -0,0 +1,97 @@
+// main package include the entry point of the todo api application
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"todo-api-golang/internal/config"
+	"todo-api-golang/internal/platform/mongo"
+	"todo-api-golang/internal/ratelimit"
+	"todo-api-golang/internal/todo"
+	"todo-api-golang/pkg/logs"
+
+	"syscall"
+	"time"
+
+	"github.com/gorilla/handlers"
+	"go.uber.org/zap"
+)
+
+// main is the entry point of the todo rest api.
+func main() {
+	logs, err := logs.New()
+	if err != nil {
+		log.Fatalf("Error initializing zap: %s\n", err)
+	}
+
+	config, err := config.LoadConfig("./../..")
+	if err != nil {
+		logs.Logger.Fatal("Cannot load config", zap.String("detail", err.Error()))
+	}
+
+	startHTTPServer(config, logs)
+}
+
+// startHTTP server initialize the server http.
+func startHTTPServer(config *config.Config, logs *logs.Logs) {
```

Comment:
I think we can move this function to a separate file or even to `internal/todo` just to keep the entrypoint of this app cleaner 😊

Reply:
Yeah for sure, everything can be refactored as required, not only this particular function but also the architecture of the project, I'm not sure which project layout suits you best. N layer / vertical slice / clean / package oriented design, so I will adapt to the standards you are using :metal:

Reply:
✔️ Done!